packages/flutter/lib/src/gestures/converter.dart

```diff
@@ -283,6 +283,9 @@ abstract final class PointerEventConverter {
                 position: position,
                 scrollDelta: scrollDelta,
                 embedderId: datum.embedderId,
+                whenIgnored: () {
```

Comment:
This API allows an asynchronous gap between the time the engine pinged the framework about the event, and the time `whenIgnored` is called. I'm not sure if there's a better way to structure the API though. Maybe this can be watched using `assert`s.

Reply:
Similar question: Does the engine need a synchronous response? Or would an async response be ok? If it needs to be sync, can we guarantee that? As I pointed out in https://github.com/flutter/flutter/issues/139263#issuecomment-1876133629 there appear to be some situations where asynchronous gaps may be present in the pointer processing pipeline (although this needs further investigation).

Reply:
Also: a synchronous responds will certainly not be possible for more complex gestures, where the framework needs to see a stream of events before deciding whether they are going to be handled or not.

Reply:
> Does the engine need a synchronous response?

Yes, it does, any asynchronicity means that the event is lost, see this super simple example: https://jsfiddle.net/ditman/z0u374yj

> synchronous responses will certainly not be possible for more complex gestures, where the framework needs to see a stream of events before deciding whether they are going to be handled or not.

I think that may be OK, as long as we eventually stop the stream, the engine would start letting the browser see some of those events. I'm not sure if that'd be enough to fix drag-to-scroll on mobile, for example, but it's worth a try.

The way I'd try to do this with gestures (not signals) is that once a packet from a gesture gets ignored, I wouldn't send any more packets to the framework until some other event "resets" the gesture (like "pointerup"). This is similar to what `setPointerCapture` does ([docs](https://developer.mozilla.org/en-US/docs/Web/API/Element/setPointerCapture))

It'd be much better if we knew beforehand if the engine is going to handle an event or not, but I'm afraid it's not possible, because at the engine we don't even have the concept of "drag", just pointer down/up/move and a few other primitives.